### DISCUSSION TITLE
Empty unpublish permission still defaults to publish

### DISCUSSION THREAD

**Discussion by ASDFGerte** [verdaccio/verdaccio#3004] [Q&A]
When specifying an empty unpublish group, permissions default back to publish permissions anyways, unlike what the documentation implies. Adding an invalid username as a workaround is possible.

verdaccio 5.6.0, no extra plugins installed

Relevant parts of config:
```text
packages:
  '@myuser/*':
    access: access myuser
    publish: myuser
    unpublish:

  '**':
    access:
    publish:
    unpublish:
    proxy:
```

I am authenticating as "myuser", and it allows unpublish, because it falls back to publish permissions:
```text
auth/allow_action: access granted to: undefined
info --- myuser is allowed publish for @myuser/somepackage
```
(note that the undefined itself is odd, but doesn't appear to be a problem - i don't understand why it's there, but it doesn't seem to have the last say when it comes to permissions) 

`npm unpublish @myuser/somepackage --registry=https://someip/ --cafile=cert.crt --force`

---

From what i can see, in the function returned by [`handleSpecialUnpublish`](https://github.com/verdaccio/verdaccio/blob/13310814da60e7f9113347f1ab5529dcc10fd80d/packages/auth/src/utils.ts#L220), there are two checks first.

One for for `_.isNil(pkg[action])` (which can imho never happen, as `pkg: AuthPackageAllow`, `const action = 'unpublish'`, and `interface AuthPackageAllow extends PackageAccess, AllowAccess {}`, `interface PackageAccess { /*...*/ unpublish: string[] }`, and `strict: true` is enabled in the compiler-settings, so neither `undefined` nor `null` are in the type - which is slightly confusing to me, as in yaml, `unpublish:` by itself would parse to `{ unpublish: null }`, but i didn't look at how the config is parsed).

Another for `(pkg[action] as string[]).length > 0`, which sanely can't happen, if nothing was ever specified.

It therefore will from what i can see always `return callback(null, undefined);`, which [defaults to permissions for `publish`](https://github.com/verdaccio/verdaccio/blob/18dc5f1f2fd97c376e75799a4b4c3959473ab26a/packages/auth/src/auth.ts#L330).

This is contrary to [what is written in the docs](https://verdaccio.org/docs/packages/#unpublishing-packages), where an empty `unpublish:` is shown.

However, disclaimer again: i only started reading verdaccio code less than a day ago, so most of the codebase and its inner workings are still a mystery to me. I just searched, and made assumptions i consider permissible.

PS: i did not post this as a bug report, as i am moderately new to verdaccio, and the related template specifically said to open a discussion instead, if there was any uncertainty.

**Comment by edclement** [comment]
Based on the information you provided and looking at the code I would agree this might be a bug. I'd want to fire up Verdaccio with a debugger attached and walk through the call stack before saying it definitely is. You might be able to get around it by:
```yaml
unpublish: ''
```

...since the [`isNil`](https://docs-lodash.com/v4/is-nil/) function only returns true if the value is `null` or `undefined`.
